lib/modules/csv_map_parser.rb

```diff
@@ -22,23 +40,44 @@ def ch3_map1_percentage
 
   private
   
-  def categ_country_stats(file_name)
-    csv_file = file_reader(file_name)
+  def categ_country_stats(config)
+    csv_file = file_reader(config[:csv])
     data = []
+
     CSV.parse(csv_file, headers: true) do |row|
       iso3 = row[0]
-      categ_stats = row.to_hash.select{|s| s.include?('categorical')}.values.first
-      data.append([iso3, categ_stats])
+      categ_stat = category(row.to_hash, config)
+      
+      data.append([iso3, categ_stat])
     end
+
     data
   end
 
+  def category(row_hash, config)
+    category = 0
+
+    if config.key?(:boundaries)
+      percentage = row_hash.select{|s| s == config[:header_name]}.values.first
+
+      config[:boundaries].each do |boundary|
+        if percentage.to_f > boundary then category += 1 end
```

Comment:
It's very unusual to see an `if then` clause in ruby.
This is because you can write the same as, for example: 
`category += 1 if percentage.to_f > boundary`

Reply:
Gotcha - will fix